backend/tests/test_add_item.py

```diff
@@ -89,6 +88,102 @@ def test_add_new_item_with_image(client):
     assert new_price.downvotes == []
 
 
+def test_add_new_item_with_image(client, db):
+    nonexistent_item = test_data.item13
+
+    name = str(nonexistent_item.name)
+    upc = str(nonexistent_item.upc)
+    image = str(base64.b64encode(nonexistent_item.image))
+
+    store = nonexistent_item.stores[0]
+    store_name = str(store.name)
+    lat = float(store.location['lat'])
+    long_arg = float(store.location['long'])
+
+    price_data = store.prices[0]
+    price = float(price_data.price)
+    user = str(price_data.user)
+
+    rv = client.post('/item', data=json.dumps({
+        'name': name,
+        'upc': upc,
+        'price': price,
+        'user': user,
+        'store': store_name,
+        'lat': lat,
+        'long': long_arg,
+        'image': image
+    }))
+    response = json.loads(rv.data)
+    assert response == {'success': True, 'error': None}
+
+    new_item = model.Item.objects(
+        upc=upc).first()
+    assert str(new_item.name) == name
+    new_store = new_item.stores[0]
+    assert str(new_store.name) == store_name
+    assert float(new_store.location['lat']) == lat
+    assert float(new_store.location['long']) == long_arg
+    new_price = new_store.prices[0]
+    assert float(new_price.price) == price
+    assert new_price.upvotes == []
+    assert new_price.downvotes == []
+
+    assert new_item.image is not None
+    # Check substring in image_url
+    assert new_item.image_url.find('/get_image?upc=' + new_item.upc) != -1
+
+
+def test_add_new_item_image_over_url(client, db):
+    '''
+    Test that the image field has priority over image_url
+    '''
+    nonexistent_item = test_data.item13
```

Comment:
I believe you use nonexistent_item to refer to an item that has yet to be added. However, this item does exist, just not yet in the db and so a better name may be new_item.

Reply:
Good point. The original reason was because one of our test fixtures is named nonexistent_item (some tests never add the item from this fixture, so calling the fixture new_item would be misleading). But you're right that the name is misleading in some tests, so we made the following changes:

In the tests that use the nonexistent_item fixture and then add it, we made a new variable new_item = nonexistent_item. For tests that don't use the fixture, we simply renamed nonexistent_item to new_item.